### ISSUE TITLE
entire backend API Proxy feature

### ISSUE THREAD

**Issue by TylerShin** [danielcondemarin/serverless-next.js#258]
When using serverless-next-js with Serverless Component, there isn't input for api gateway setting.

For example, when using the legacy serverless.yml, we can override current default api gateway like below example. 

```
resources:
  Resources:    
    APIProxyResourceParent:
      Type: AWS::ApiGateway::Resource
      Properties:
        ParentId:
          Fn::GetAtt:
            - ApiGatewayRestApi
            - RootResourceId
        PathPart: api
        RestApiId:
          Ref: ApiGatewayRestApi
    APIProxyResource:
      Type: AWS::ApiGateway::Resource
      Properties:
        ParentId:
          Ref: APIProxyResourceParent
        PathPart: '{proxy+}'
        RestApiId:
          Ref: ApiGatewayRestApi
    APIProxyMethod:
      Type: AWS::ApiGateway::Method
      Properties:
        ResourceId:
          Ref: APIProxyResource
        RestApiId:
          Ref: ApiGatewayRestApi
        HttpMethod: ANY
        AuthorizationType: NONE
        RequestParameters:
         method.request.path.proxy: true
        Integration:
          IntegrationHttpMethod: any
          Type: HTTP_PROXY
          Uri: ${file(./env/${opt:stage}.yml):API_ENDPOINT}
          RequestParameters:
           integration.request.path.proxy: method.request.path.proxy
          IntegrationResponses:
            - StatusCode: 200
```

I know that I can handle each endpoint of API gateway inside of NextJS from v9. however, making all of the API endpoint seems to redundant and annoying.

Also NextJS official document saying we should implement these proxy behavior by making custom server(sth like express or custom routing for Now)

However, we couldn't use custom server with serverless-next.js.
So I think this feature should be handled by serverless-next.js.

is there any solution for the proxy feature?

**Comment by danielcondemarin** [comment]
I believe this could be a feature made available via @barrysteyn 's proposal in https://github.com/danielcondemarin/serverless-next.js/issues/270. 

I'm thinking of something like:

```yml
component: serverless-next.js
inputs:
 cloudfront: 
   origins:
     url: https://somebackend.com
     pathPatterns:
        /this/path/is/proxied: 
         ttl: 5
         ... # other inputs in https://github.com/serverless-components/aws-cloudfront
```

This allows for greater flexibility in the CloudFront distribution provisioned by the component.

**Comment by TylorShin** [comment]
> I believe this could be a feature made available via @barrysteyn 's proposal in #270.
> 
> I'm thinking of something like:
> 
> ```yaml
> component: serverless-next.js
> inputs:
>  cloudfront: 
>    origins:
>      url: https://somebackend.com
>      pathPatterns:
>         /this/path/is/proxied: 
>          ttl: 5
>          ... # other inputs in https://github.com/serverless-components/aws-cloudfront
> ```
> 
> This allows for greater flexibility in the CloudFront distribution provisioned by the component.

awesome

**Comment by danielcondemarin** [comment]
Closing, this was implemented in https://github.com/danielcondemarin/serverless-next.js/pull/376
